### PR TITLE
Phylocanvas as a BioJS component for the Galaxy charts

### PR DESCRIPTION
**Pull request by anuprulez**
# Introduction:

This PR introduces [Phylocanvas](http://phylocanvas.org/) as a BioJS component for the Galaxy charts to visualize phylogenetic/ evolutionary trees which themselves are pivotal to understand ancestral relationships among innumerable species of life forms. Phylocanvas offers an interactive view of the phylogenetic trees.

# First look:

From this library, we include the following features:

- Choose among five different tree types - Circular, Diagonal, Hierarchical, Radial and Rectangular.
- Select different colors for trees, highlighted and selected sections of a tree.
- Collapse and prune part(s) of a tree.
- Show/hide and align labels of the leaves of a tree.
- Select different shapes of leaves of a tree.

Collapsed part is dark shaded:

Circular tree type:

# Data format:
The text files containing data in [Newick](https://en.wikipedia.org/wiki/Newick_format) (tree) notation can be used to visualize the trees. Few data files to go through the features:

[file1.txt](https://github.com/galaxyproject/galaxy/files/883644/file1.txt)
[file2.txt](https://github.com/galaxyproject/galaxy/files/883643/file2.txt)
[file3.txt](https://github.com/galaxyproject/galaxy/files/883645/file3.txt)
[file4.txt](https://github.com/galaxyproject/galaxy/files/883646/file4.txt)
[file5.txt](https://github.com/galaxyproject/galaxy/files/883682/file5.txt)
[file7.txt](https://github.com/galaxyproject/galaxy/files/883681/file7.txt)

# Courtesy:
Phylocanvas is maintained by Centre for Genomic Pathogen Surveillance ([CGPS](http://www.pathogensurveillance.net/))

# Credits:
Thanks a lot to Aysam Guerler (@guerler) and Björn Grüning (@bgruening) for their suggestions and review.

# Note:
While reviewing, we found a missing feature in the library (for Radial tree type) and opened a feature request [here](https://github.com/phylocanvas/phylocanvas/issues/210).

All suggestions are welcome. Thanks a lot!

### CODE REVIEW

**Comment by bgruening** [comment]
It's getting better and better :)
Thanks a lot @anuprulez!

File: config/plugins/visualizations/charts/static/repository/visualizations/biojs/phylocanvas/config.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+define( [], function() {
+    return {
+        title       : 'Phylogenetic Tree Visualization',
+        library     : 'BioJS',
+        datatypes   : [ 'txt', 'nwk' ],
```
Comment (nsoranzo) [inline config/plugins/visualizations/charts/static/repository/visualizations/biojs/phylocanvas/config.js]: If these are Galaxy datatypes, then the only correct one is `nhx`.

**Comment by gregvonkuster** [comment]
@anuprulez Wow!  This is awesome!  Thanks so much for this - this will be so beneficial some of what I'm involved in!  ;)

**Comment by bgruening** [comment]
@nsoranzo is this ok now?

**Comment by nsoranzo** [comment]
Yes, thanks.

**Comment by anuprulez** [comment]
@bgruening @guerler I merged this branch with the dev, can this PR be merged? Thanks!

**Comment by guerler** [comment]
Can you rebase/undo the merge and make one clean merge? I am not sure how to check if this merge is ok and would like to avoid a bad merge since those are very difficult to resolve down the road.

**Comment by anuprulez** [comment]
@guerler I reverted the last two commits and did one clean merge with dev.

**Comment by anuprulez** [comment]
@guerler Thanks a lot for merging :)

**Pull request merged**